azuredevops/internal/service/git/resource_git_repository.go

```diff
@@ -394,9 +394,14 @@ func expandGitRepository(d *schema.ResourceData) (*git.GitRepository, *repoIniti
 
 	var initialization *repoInitializationMeta = nil
 	initData := d.Get("initialization").(*schema.Set).List()
-
-	// Note: If configured, this will be of length 1 based on the schema definition above.
-	if len(initData) == 1 {
+	if len(initData) == 0 {
```

Comment:
@xuzhang3 @tmeckel Should initialization be a required field? I can't find a example where it's not set to be something?

Reply:
Not required. But if not set, terraform will crash #54

Reply:
This PR should fix that and default to `Uninitialised`. I think `Uninitialised` makes sense as a default, as the repo will be created with the default branch and no commits.

Reply:
@beandrad Instead initializing it with default, let's make the `initialization` field required with 

```go
			"initialization": {
				Type:     schema.TypeSet,
				Required: true,
				MaxItems: 1,
```

Reply:
I agree with @EliiseS , better force people to be explicit on what they want, instead for heading with a (potentially) unwanted default.